### DISCUSSION TITLE
Divide the SSH runner into SSH+password, SSH+OpenSSH, and SSH+putty PPK.

### DISCUSSION THREAD

**Discussion by VShawn** [1Remote/1Remote#426] [Ideas]
would be good to have automatically select different SSH runners when users input different types of keys. 

Especially different SSH CLI tools support different inputs: some only support passwords, some only support putty PPK, and some only support OpenSSH private keys.

**Comment by smaiiboy** [comment]
是否可以考虑集成密钥转换工具？我迁移Xshell的时候发现，必须下载putty转换私钥，而且不支持私钥密码。（Store版本）

**Reply by VShawn** [reply]
我没用过openssh 的密钥，问一下openssh私钥转成putty私钥后，还需要在服务器端重新配置putty私钥对应的公钥么？还是直接用转出来的私钥就能打开SSH？

**Reply by smaiiboy** [reply]
测试了一下，经过Putty Key Generator转换成ppk的私钥，可以直接使用本工具连接（服务器为server2008，安装的openssh）。因为密钥格式不同，同时打开的SFTP不能连接，需要用openssh私钥单独配置一个

**Reply by VShawn** [reply]
https://www.chiark.greenend.org.uk/~sgtatham/putty/wishlist/puttygen-batch.html

因为 puttygen 缺少 CLI，所以我们无法做到后台自动将open ssh的密钥静默转为putty key。
